File: output/anthropic_claude_sonnet_4_6/few/python/Torre.py
import sys

data = sys.stdin.read().split()
idx = 0
n = int(data[idx]); idx += 1

grid = []
for i in range(n):
    row = [int(data[idx+j]) for j in range(n)]
    idx += n
    grid.append(row)

row_sum = [sum(grid[i]) for i in range(n)]
col_sum = [sum(grid[i][j] for i in range(n)) for j in range(n)]

max_weight = 0
for i in range(n):
    for j in range(n):
        weight = row_sum[i] + col_sum[j] - grid[i][j]
        if weight > max_weight:
            max_weight = weight

print(max_weight)

File: output/csbc_semish_2026/x_ai_grok_code_fast_1/zero/cpp/Torre.cpp

#include <bits/stdc++.h>
using namespace std;

int main() {
    int N;
    cin >> N;
    vector<vector<int>> grid(N, vector<int>(N));
    for(int i = 0; i < N; i++) {
        for(int j = 0; j < N; j++) {
            cin >> grid[i][j];
        }
    }
    vector<long long> row_sum(N, 0);
    vector<long long> col_sum(N, 0);
    for(int i = 0; i < N; i++) {
        for(int j = 0; j < N; j++) {
            row_sum[i] += grid[i][j];
            col_sum[j] += grid[i][j];
        }
    }
    long long max_weight = 0;
    for(int i = 0; i < N; i++) {
        for(int j = 0; j < N; j++) {
            long long weight = row_sum[i] + col_sum[j] - 2LL * grid[i][j];
            if(weight > max_weight) max_weight = weight;
        }
    }
    cout << max_weight << endl;
    return 0;
}
